Source/OxyPlot.SkiaSharp.Tests/OxyPlot.SkiaSharp.Tests.csproj

```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <TargetFrameworks>net461;netcoreapp3.1</TargetFrameworks>
+        <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+        <Description>OxyPlot SkiaSharp unit tests</Description>
+        <PackageLicenseExpression>MIT</PackageLicenseExpression>
+        <Copyright>OxyPlot contributors</Copyright>
+        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+        <Platforms>AnyCPU;x86;x64</Platforms>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(Platform)'=='AnyCPU'">
+        <PlatformTarget>x86</PlatformTarget>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(Platform)'=='x86'">
+        <PlatformTarget>x86</PlatformTarget>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(Platform)'=='x64'">
+        <PlatformTarget>x64</PlatformTarget>
+    </PropertyGroup>
```

Comment:
For reasons I do not understand, I had to change this further to

    <PropertyGroup Condition="'$(TargetFramework)|$(Platform)'=='net461|AnyCPU'">
        <PlatformTarget>x86</PlatformTarget>
    </PropertyGroup>
    <PropertyGroup Condition="'$(TargetFramework)|$(Platform)'=='net461|x86'">
        <PlatformTarget>x86</PlatformTarget>
    </PropertyGroup>
    <PropertyGroup Condition="'$(TargetFramework)|$(Platform)'=='net461|x64'">
        <PlatformTarget>x64</PlatformTarget>
    </PropertyGroup>

in order to get the tests to run on my WSL Ubuntu instance (i.e. don't set PlatformTarget under .NET Core; doesn't seem to be a problem on my windows box).

Reply:
What was the issue otherwise? Did you get a `BadImageFormatException` or something else?

Reply:
Not really sure, never really needed to understand NUnit. Here's the [output in a gist](https://gist.github.com/VisualMelon/3ae85794600a8167c40c3ccf1d4b9774).

It reports a `BadImageFormatException` on the Tests file itself: the error seems rather uncharacteristic to be honest, and I'm not sure what is going on; I was under the impression that `PlatformTarget` was ignored by net core.

Reply:
Update: It works when `PlatformTarget` is set to x64.

Reply:
Sorry for all the confusion. I've been using .NET Core for years without realising that you can target platforms without publishing. The default is AnyCPU, and I believe there is no reason to have otherwise here.

Now I just need to work out why the x86 version targeting NetCore 3.1 runs on my Windows box when I don't have an x86 version of NetCore 3.1...

Reply:
I'm similarly baffled as you that I was oblivious to this behavior of .NET Core until now. I'm still not completely sure I fully understand what is going on, but I think it is like this:

- .NET Core ignores the `PlatformTarget` and creates a cross-platform app that can run on all supported architectures and operating systems
- Depending on the environment the app is executed as either x86 or x64
- The test host is deployed as either x86 or x64 (but not both of them)
- When there is a mismatch between the test host and the (runtime) app architecture you get the `BadImageFormatException`

I think for the specific case of the Test project, the most sensible way is to force one specific architecture on both frameworks, for example x64. So we can be sure the x64 test host is deployed and has a x64 app to work with. To do that I think we should specify the `RuntimeIdentifier` for netcoreapp3.1. So we end up with the project file as in this gist: 
https://gist.github.com/Jonarw/f98dde40a50dbd8a30faf3dd9b11b85b

Do you agree? Did you come to the same conclusions as me? Or did I misunderstand something?

Reply:
It is irritating how little documentation there is about this. Based on my observations, my new understanding is this:
 - `PlatformTarget` is not ignored by .NET Core, and defaults to `AnyCPU`. Seems to have the same behaviour with .NET Framework
 - For `AnyCPU`, this means the runtime is determined by the host: if you run it will NET Core x64, it will run as x64; x86 for an x86 version of NET Core. In this case, it is able to pick the right native DLLs because it ships both
 - My Linux run failed because the current csproj was default to x86, because I wasn't asking for anything else. It worked when I unconditionally set `PlatformTarget` to `AnyCPU` or `x64`, because the x64 host can deal with either of these.
 - My Windows runs worked because I had an x86 .NET Core 3.1 host installed, a fact I didn't realise at the time. (I thought it was working because I thought `PlatformTarget` was ignored under core)

Because the image type is determined by the host, I think AnyCPU would be the best default for .NET Core.

Seeing as all the CIs are Windows, I don't think it's the biggest issue in the world (and I think this should be merged as soon as possible; this is the sort of thing that won't influence the design which we can resolve later). I would, however, prefer that it wasn't unconditionally be x64 given x86 should (and does) work: which is used for CI doesn't seem to matter. I'd also prefer the that the NET Core config wasn't windows specific in any case.

By luck more than anything else, I think my most recent suggestion would be the best option, as it allows the platform to be qualified under framework, defaulting to x86 which is the better-supported image format. It doesn't touch the config for .NET Core, defaulting to AnyCPU (which should work anywhere), and not interfering with anything that might be specified by a user. Changing the `AnyCPU` override to `x64` would equally be fine.

But again, these are just tests, so maybe I'm wasting my effort worrying about it. I've certainly learned a thing or two in the last couple of days... or perhaps you can tell me what I've got wrong ;)

Reply:
> It is irritating how little documentation there is about this.
Yes, I was thinking that as well...

What you say sounds plausible, and I think is also consistent with my observations.

> But again, these are just tests, so maybe I'm wasting my effort worrying about it. I've certainly learned a thing or two in the last couple of days...

Yes indeed, me as well.